tests/consensus_tests/test_collection_shard_update.py

```diff
@@ -134,5 +134,5 @@ def test_collection_shard_update(tmp_path: pathlib.Path):
     assert r.status_code == 400
     error = r.json()["status"]["error"]
     assert error.__contains__("Wrong input: 1 out of 2 shards failed to apply operation")
-    assert error.__contains__("Wrong input: InvalidArgument")
+    assert error.__contains__("Wrong input: Vector inserting error: expected dim")
```

Comment:
What is going on here, did the error changed?

Reply:
Yes. Prior to this PR, updates were first replicated to remote shards, and then applied to the local shard. I've changed it to "local first, then remote". The error message is different for local and remote updates.

However, I now realized I might have introduced some flakiness to this: the order in which errors are returned are unspecified now, so it might be either the new message, or the old one. I'll change it to `assert error.__contains__("Wrong input: InvalidArgument") or error.__contains__("Wrong input: Vector inserting error: expected dim")` in a separate PR later today.